specs/Extensions.md

```diff
@@ -0,0 +1,585 @@
+/// includes the number of browser Extensions in the list, and the ability
+/// to get an browser Extension from the list at a particular index.
+[uuid(59251055-F2F1-448F-A096-F996FB9ACBE2), object, pointer_default(unique)]
+interface ICoreWebView2BrowserExtensionList : IUnknown {
+    /// The number of browser Extensions in the list.
+    [propget] HRESULT Count([out, retval] UINT* count);
+    /// Gets the browser Extension located in the browser Extension List at the given index.
+    HRESULT GetValueAtIndex([in] UINT index, [out, retval] ICoreWebView2BrowserExtension** extension);
+}
+
+/// The caller implements this interface to receive the result of
+/// getting the browser Extensions.
+[uuid(2A05565F-DE9D-46E3-AAF7-B866966828F4), object, pointer_default(unique)]
+interface ICoreWebView2ProfileGetBrowserExtensionsCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2BrowserExtensionList* extensionList);
+}
+
+/// The caller implements this interface to receive the result
+/// of loading an browser Extension.
+[uuid(E81499FE-8BC6-4BA6-BAD7-D21FFA4C3266), object, pointer_default(unique)]
+interface ICoreWebView2ProfileAddBrowserExtensionCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2BrowserExtension* extension);
+}
+
+/// The caller implements this interface to receive the result of removing
+/// the browser Extension from the Profile.
+[uuid(BD73ED6B-08A3-4A57-9DD0-2903067632B4), object, pointer_default(unique)]
+interface ICoreWebView2BrowserExtensionRemoveCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// The caller implements this interface to receive the result of setting the
+/// browser Extension as enabled or disabled. If enabled, the browser Extension is 
+/// running in WebView instances. If disabled, the browser Extension is not running in WebView instances.
+[uuid(0A5F7098-2265-49C7-BA60-5A511E80805A), object, pointer_default(unique)]
+interface ICoreWebView2BrowserExtensionSetEnabledCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    
+    // ...
+    runtimeclass CoreWebView2BrowserExtension;
+    
+    // ...
+    runtimeclass CoreWebView2EnvironmentOptions
+    {
+        // ...
+        Boolean AreBrowserExtensionsEnabled { get; set; };
+    }
+
+    // ...
+    runtimeclass CoreWebView2Profile
+    {
+        // ...
+        Windows.Foundation.IAsyncOperation<CoreWebView2BrowserExtension> AddBrowserExtensionAsync(String extensionFolderPath);
+        IAsyncOperation<IVectorView<CoreWebView2BrowserExtension>> GetBrowserExtensionsAsync();
+    }
+
+    runtimeclass CoreWebView2BrowserExtension
+    {
+        String Id { get; };
+        String Name { get; };
+        Boolean IsEnabled { get; };
+        Windows.Foundation.IAsyncAction RemoveAsync();
```

Comment:
Why the asymmetry adding extensions on CoreWebView2Profile but removing the with an instance method?  Would this be easier to use in practice than supporting "Windows.Foundation.IAsyncAction RemoveExtensionAsync(CoreWebView2BrowserExtension extension);" on CoreWebView2Profile?